src/test/mir-opt/intrinsic_asserts.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(core_intrinsics)]
+
+// EMIT_MIR intrinsic_asserts.main.InstCombine.diff
+fn main() {
```

Comment:
Could you add a variant of this test where the intrinsics are called with a type param? To verify that they do not disappear in that case.